### PR TITLE
squirrel-sql: Add MSSQL driver.

### PR DESCRIPTION
**Pull request by cresh**
###### Motivation for this change

Use squirrel-sql to talk to MSSQL databases.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by aanderse** [comment]
@cresh I've never used squirrel-sql before but I checkout out your branch, selected "Microsoft MSSQL Server JDBC Driver" and then tried to connect to a mssql database but got the following error:

> troy: JDBC Driver class not found
class java.lang.ClassNotFoundException: com.microsoft.sqlserver.jdbc.SQLServerDriver

Please advise.

**Comment by cresh** [comment]
@aanderse I've just fixed a regression, the set CLASS wasn't picked up correctly. Please give it another spin, it should work now.

File: pkgs/servers/sql/mssql/jdbc/default.nix [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{stdenv, fetchurl, ant, unzip}:
+
+stdenv.mkDerivation rec {
+  name = "mssql-jdbc-7.2.0.jre8";
```
[the remark this answers is not in the document]
Comment (aanderse) [reply pkgs/servers/sql/mssql/jdbc/default.nix]: You also might consider using `version` and `pname` to reduce typing and chance of mistake, like so:
```
stdenv.mkDerivation rec {
  pname = "mssql-jdbc";
  version = "7.2.0";
  builder = ./builder.sh;

  src = fetchurl {
    url = "https://github.com/Microsoft/mssql-jdbc/releases/download/v${version}/${pname}-${version}.jre8.jar";
    sha256 = "1552brlwk7471ngmr1m08cjxl7bxascgrsmvrnq7d3kj4i9z7j96";
  };
```

**Review by aanderse — CHANGES_REQUESTED** [review]
Noticed a few things this time around.

**Comment by aanderse** [comment]
> @aanderse I've just fixed a regression, the set CLASS wasn't picked up correctly. Please give it another spin, it should work now.

It worked now, thanks!

**Comment by cresh** [comment]
Thanks for the feedback! I've just made the requested changes.

**Review by aanderse — APPROVED** [review]

**Comment by aanderse** [comment]
@GrahamcOfBorg build squirrel-sql

**Comment by aanderse** [comment]
Thanks for your contribution @cresh! I'll probably make use of this at some point :smiley:

**Pull request merged**